src/main/java/org/prebid/server/bidder/dmx/DmxBidder.java

```diff
@@ -210,7 +208,15 @@ private Imp fetchParams(Imp imp, ExtImpDmx extImp) {
                     .build();
         }
 
-        return (updatedImp != null && StringUtils.isBlank(updatedImp.getTagid())) ? imp : updatedImp;
+        return updatedImp != null && StringUtils.isBlank(updatedImp.getTagid()) && StringUtils.isBlank(imp.getTagid())
+                ? null
+                : updatedImp;
+    }
+
+    private void checkIfHasId(boolean anyHasId) {
```

Comment:
Not that. 
I mean that we can extract **ALL** related logic to this variable to separate method